copy.go

```diff
@@ -262,7 +274,10 @@ func dcopyConcurrent(srcdir, destdir string, contents []os.FileInfo, opt Options
 
 func onDirExists(opt Options, srcdir, destdir string) (bool, error) {
 	_, err := os.Stat(destdir)
-	if err == nil && opt.OnDirExists != nil && destdir != opt.intent.dest {
+	if err != nil && !os.IsNotExist(err) {
+		return true, err // Unwelcome error type...!
+	}
+	if opt.OnDirExists != nil && destdir != opt.intent.dest {
 		switch opt.OnDirExists(srcdir, destdir) {
```

Comment:
IMO, we can separate another exception case `destdir == opt.intent.dest` on the top of this function for early-return.